dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -405,14 +402,92 @@ public sealed class DidDhtApi(configuration: DidDhtConfiguration) : DidMethod<Di
       val typeIndexes = types.map { it.index }
       message.addRecord(
         TXTRecord(
-          Name("_typ._did."), DClass.IN, ttl, "id=${typeIndexes.joinToString(",")}"
+          Name("_typ._did."), DClass.IN, ttl, "id=${typeIndexes.joinToString(ARRAY_SEPARATOR)}"
         ), Section.ANSWER
       )
     }
 
     return message
   }
 
+  private fun addVerificationMethodRecords(didDocument: DIDDocument, message: Message):
+    Pair<List<String>, Map<String, String>> {
+    val verificationMethodsById = mutableMapOf<String, String>()
+    val verificationMethods = buildList {
+      didDocument.verificationMethods?.forEachIndexed { i, verificationMethod ->
+        val publicKeyJwk = JWK.parse(verificationMethod.publicKeyJwk)
+        val publicKeyBytes = Crypto.publicKeyToBytes(publicKeyJwk)
+        val base64UrlEncodedKey = Convert(publicKeyBytes).toBase64Url(padding = false)
+        val verificationMethodId = "k$i"
+
+        verificationMethodsById[verificationMethod.id.toString()] = verificationMethodId
+
+        val keyType = when (publicKeyJwk.algorithm) {
+          JWSAlgorithm.EdDSA -> 0
+          JWSAlgorithm.ES256K -> 1
+          else -> throw IllegalArgumentException("unsupported algorithm: ${publicKeyJwk.algorithm}")
+        }
+
+        message.addRecord(
+          TXTRecord(
+            Name("_$verificationMethodId._did."),
+            DClass.IN,
+            ttl,
+            buildList {
+              add("id=${verificationMethod.id.rawFragment}")
+              add("t=$keyType")
+              add("k=$base64UrlEncodedKey")
+              if (verificationMethod.jsonObject.containsKey("controller")) {
+                add("c=${verificationMethod.jsonObject["controller"]}")
+              }
+            }.joinToString(PROPERTY_SEPARATOR)
+          ), Section.ANSWER
+        )
+
+        add(verificationMethodId)
+      }
+    }
+    return Pair(verificationMethods, verificationMethodsById)
+  }
+
+  private fun addAlsoKnownAsRecord(didDocument: DIDDocument, message: Message) {
+    if (didDocument.alsoKnownAses.isNullOrEmpty()) {
+      return
+    }
+    message.addRecord(
+      TXTRecord(
+        Name("_aka._did."),
+        DClass.IN,
+        ttl,
+        didDocument.alsoKnownAses.joinToString(PROPERTY_SEPARATOR)
```

Comment:
if this is a single value what happens?

Reply:
didDocument.alsoKnownAses always returns a list. 

Join returns the the value of the single element when the list is of size 1.